components/style/servo/selector_parser.rs

```diff
@@ -100,44 +100,55 @@ impl PseudoElement {
 #[cfg_attr(feature = "servo", derive(HeapSizeOf))]
 #[allow(missing_docs)]
 pub enum NonTSPseudoClass {
-    AnyLink,
-    Link,
-    Visited,
     Active,
+    AnyLink,
+    Checked,
+    Disabled,
+    Enabled,
     Focus,
     Fullscreen,
     Hover,
-    Enabled,
-    Disabled,
-    Checked,
     Indeterminate,
-    ServoNonZeroBorder,
+    Lang(String),
+    Link,
+    PlaceholderShown,
     ReadWrite,
     ReadOnly,
-    PlaceholderShown,
+    ServoNonZeroBorder,
     Target,
+    Visited,
 }
 
 impl ToCss for NonTSPseudoClass {
     fn to_css<W>(&self, dest: &mut W) -> fmt::Result where W: fmt::Write {
         use self::NonTSPseudoClass::*;
+        if let Lang(ref lang) = *self {
+            dest.write_str(":lang(")?;
+            serialize_identifier(lang, dest)?;
+            return dest.write_str(")");
+        }
+
         dest.write_str(match *self {
-            AnyLink => ":any-link",
-            Link => ":link",
-            Visited => ":visited",
             Active => ":active",
+            AnyLink => ":any-link",
+            Checked => ":checked",
+            Disabled => ":disabled",
+            Enabled => ":enabled",
             Focus => ":focus",
             Fullscreen => ":fullscreen",
             Hover => ":hover",
-            Enabled => ":enabled",
-            Disabled => ":disabled",
-            Checked => ":checked",
             Indeterminate => ":indeterminate",
+            Lang(_) => {
```

Comment:
But it _is_ unreachable, since we do handle it right above this match. If it reaches this, it's a bug, so I'd like it to panic.

If you feel pretty strongly about it, at least `debug_assert!(false)`, or whatever, but...